lib/jekyll/commands/serve.rb

```diff
@@ -33,7 +33,9 @@ def init_with_program(prog)
             cmd.action do |_, opts|
               opts["serving"] = true
               opts["watch"  ] = true unless opts.key?("watch")
+              cfg = opts["config"]
               Build.process(opts)
+              opts["config"] = cfg
```

Comment:
Why does this need to happen? This is not only highly disoptimized by I don't understand why it has to happen and if it does I think there is a much bigger problem that needs to be addressed that either should be addressed here or this blocked on that being fixed.